FueledUtils/ReactiveSwiftExtensions.swift

```diff
@@ -160,7 +123,66 @@ public extension SignalProtocol {
 	}
 }
 
-public extension SignalProducerProtocol {
+extension SignalProducerProtocol {
+	///
+	/// Make the `SignalProducer` output optional `Value`.
+	///
+	public func promoteOptional() -> SignalProducer<Value?, Error> {
+		return self.producer.lift { $0.promoteOptional() }
+	}
+
+	///
+	/// Make the `Signal` output optional `Value`, and prefix it with `nil`.
+	///
+	public func prefixNil() -> SignalProducer<Value?, Error> {
+		return self.producer.promoteOptional().prefix(value: nil)
+	}
+
+	///
+	/// Execute the given list of SignalProducer one after the other. If any `SignalProducer` from the array fails,
```

Comment:
What's the difference with flattening using the `concat` strategy ?

Reply:
`concat` sends each value individually, whereas this will return the resulting array of values once every producer have completed.
I tried to use `.concat` and put each values into an array, but for some reason I couldn't make it work so did that instead 🤔

Reply:
If you use `concat` followed by `reduce` or `scan`, you'll get this behavior.

Reply:
Very nice! I'll remove this one then, I'll quickly verify the behavior is as expected using this combination

Reply:
Keep me posted, I may be wrong 👌

Reply:
@notbenoit I've tried, and indeed
```swift
SignalProducer(producers).flatten(.concat).reduce([]) { $0 + [$1] }
```
is equivalent to
```swift
SignalProducer.oneByOne(producers)
```
(given the same set of producers).
Should I just get rid of this method them?

Reply:
(the difference between `reduce` and `scan` is that `scan` will also send each individual values in the resulting signal producers, whereas `result` only sends the final value)

Reply:
Sounds good, I feel like we can remove the method. And the flexiblity with the `scan` and `reduce` option adds even more value and freedom than the `oneByOne` function.

Reply:
Make sense to me, removing 😉